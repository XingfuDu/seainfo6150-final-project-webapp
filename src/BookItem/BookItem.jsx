import React, { useState } from 'react';
import HTMLText from '../HTMLText/HTMLText';
import classes from './BookItem.module.css';
import BookImage from '../BookImage/BookImage';
import Button from '../Button/Button';
import ContactData from '../ContactData/ContactData';
import { Link } from 'react-router-dom';

const BookItem = (props) => {
	const [ isFormShowing, setIsFormShowing ] = useState(false);
	const [QuantityState, setQuantityState] = useState({
		Quantity: 1
	  });

	function onClick() {
		setIsFormShowing(!isFormShowing);
	}

	const description = (
		<div>
			<h3 className={classes.h3}>Overview:</h3>
			<HTMLText className={classes.description} text={props.book.description} />
		</div>
	);
	const QuantityChangedHandler = (event) => {
		setQuantityState( {
			Quantity: event.target.value
		} )
	}
	return (
		<div className={classes.container}>
			<article className={classes.article}>
				<div className={classes.content}>
					<BookImage url={props.book.image} title={props.book.title} />
					<div className={classes.wrapper}>
						<h3 className={classes.h3}>{props.book.title}</h3>
						<time className={classes.time} dateTime={props.book.timeStamp}>
							{props.book.displayDate}
						</time>
						<p className={classes.p}>
							By {props.book.author} &nbsp;&nbsp; Category:{' '}
							<Link className={classes.link} to={`/book/${props.book.categoryId}`}>
								{props.book.category}
							</Link>
						</p>
						<div className={classes.price}>
							<p className={classes.p}>
								Price: &nbsp; {props.book.price}&#36; &nbsp; Quantity:
								<input className={classes.Input} type="number" min="1" onChange={QuantityChangedHandler} />
							</p>
						</div>
						<Button btnType="Success" clicked={onClick}>
							Buy Now
						</Button>
					</div>
				</div>
				{isFormShowing ? null : description}
				{isFormShowing ? <ContactData book={QuantityState}/> : null}
			</article>
		</div>
	);
};

export default BookItem;
