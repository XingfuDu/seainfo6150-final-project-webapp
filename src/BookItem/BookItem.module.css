.container {
	background-color: #f2e9e4;
	width: 100%;
	min-height: 100%;
	font-family: Arial, Helvetica, sans-serif;
	padding: 0px;
	display: flex;
	align-items: center;
}

.article {
	min-height: 100%;
	margin-right: 0;
	font-family: Arial, Helvetica, sans-serif;
	font-size: 150%;
	width: 50%;
	padding-left: 25%;
	padding-top: 50px;
	display: flex;
	flex-direction: column;
}

.content {
	display: flex;
	flex-direction: row;
	justify-content: flex-start;
	flex-wrap: wrap;
}

.wrapper {
	flex: 1;
	padding: 0px 50px 10px;
	margin-top: 0;
	display: flex;
	flex-direction: column;
	justify-content: flex-start;
	align-items: flex-start;
}

.p,
h3, .time{
	color: #3d5a80;
}

.link {
    margin: 0 0 20px;
    color: #3d5a80;
    min-width: 50%;
}

@media (max-width: 1000px) {
	.content {
		display: flex;
		flex-direction: row;
		justify-content: center;
		flex-wrap: wrap;
	}
	.wrapper {
		align-items: center;
	}
}
