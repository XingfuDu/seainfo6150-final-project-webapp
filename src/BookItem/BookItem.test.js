import React from 'react';
import BookItem from './BookItem';
import { render } from "@testing-library/react";
import { BrowserRouter } from 'react-router-dom';

describe('BookItem tests', () => {
    const book = {
        "id": "0",
        "categoryId": "1",
        "category": "Psychology",
        "title": "Some We Love, Some We Hate, Some We Eat: Why It's So Hard to Think Straight About Animals",
        "author": "Hal Herzog",
        "description": "<p>Combining the intellect of Malcolm Gladwell with the irreverent humor of Mary Roach and the paradigm-shifting analysis of Jared Diamond, a leading social scientist offers an unprecedented look inside our complex and often paradoxical relationships with animals.</p><p>Does living with a pet really make people happier and healthier? What can we learn from biomedical research with mice? Who enjoyed a better quality of life—the chicken on a dinner plate or the rooster who died in a Saturday-night cockfight? Why is it wrong to eat the family dog? Drawing on more than two decades of research in the emerging field of anthrozoology, the science of human–animal relations, Hal Herzog offers surprising answers to these and other questions related to the moral conundrums we face day in and day out regarding the creatures with whom we share our world.</p><p>Some We Love, Some We Hate, Some We Eat is a highly entertaining and illuminating journey through the full spectrum of human–animal relations, based on Dr. Herzog’s groundbreaking research on animal rights activists, cockfighters, professional dog-show handlers, veterinary students, and biomedical researchers. Blending anthropology, behavioral economics, evolutionary psychology, and philosophy, Herzog carefully crafts a seamless narrative enriched with real-life anecdotes, scientific research, and his own sense of moral ambivalence.</p><p>Alternately poignant, challenging, and laugh-out-loud funny, this enlightening and provocative book will forever change the way we look at our relationships with other creatures and, ultimately, how we see ourselves.",
        "image": "https://i.gr-assets.com/images/S/compressed.photo.goodreads.com/books/1369453733l/6953508.jpg",
        "price": "12.99",
        "timeStamp": "2018-11-22T17:55:57.000Z",
        "displayDate": "November 22nd 2018, 9:55 am"
    };
	it('renders correctly', () => {
		const { container } = render(
			<BrowserRouter><BookItem book={book}/></BrowserRouter>
		);
		expect(container).toMatchSnapshot();
	});
});
