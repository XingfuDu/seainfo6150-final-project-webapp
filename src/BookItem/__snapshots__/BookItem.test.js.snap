// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookItem tests renders correctly 1`] = `
<div>
  <div
    class="container"
  >
    <article
      class="article"
    >
      <div
        class="content"
      >
        <img
          alt="Some We Love, Some We Hate, Some We Eat: Why It's So Hard to Think Straight About Animals"
          class="image"
          src="https://i.gr-assets.com/images/S/compressed.photo.goodreads.com/books/1369453733l/6953508.jpg"
        />
        <div
          class="wrapper"
        >
          <h3
            class="h3"
          >
            Some We Love, Some We Hate, Some We Eat: Why It's So Hard to Think Straight About Animals
          </h3>
          <time
            class="time"
            datetime="2018-11-22T17:55:57.000Z"
          >
            November 22nd 2018, 9:55 am
          </time>
          <p
            class="p"
          >
            By 
            Hal Herzog
                Category:
             
            <a
              class="link"
              href="/book/1"
            >
              Psychology
            </a>
          </p>
          <div
            class="price"
          >
            <p
              class="p"
            >
              Price:   
              12.99
              $   Quantity:
              <input
                class="Input"
                min="1"
                type="number"
              />
            </p>
          </div>
          <button
            class="Button Success"
          >
            Buy Now
          </button>
        </div>
      </div>
      <div>
        <h3
          class="h3"
        >
          Overview:
        </h3>
        <div
          class="description"
        >
          <p>
            Combining the intellect of Malcolm Gladwell with the irreverent humor of Mary Roach and the paradigm-shifting analysis of Jared Diamond, a leading social scientist offers an unprecedented look inside our complex and often paradoxical relationships with animals.
          </p>
          <p>
            Does living with a pet really make people happier and healthier? What can we learn from biomedical research with mice? Who enjoyed a better quality of life—the chicken on a dinner plate or the rooster who died in a Saturday-night cockfight? Why is it wrong to eat the family dog? Drawing on more than two decades of research in the emerging field of anthrozoology, the science of human–animal relations, Hal Herzog offers surprising answers to these and other questions related to the moral conundrums we face day in and day out regarding the creatures with whom we share our world.
          </p>
          <p>
            Some We Love, Some We Hate, Some We Eat is a highly entertaining and illuminating journey through the full spectrum of human–animal relations, based on Dr. Herzog’s groundbreaking research on animal rights activists, cockfighters, professional dog-show handlers, veterinary students, and biomedical researchers. Blending anthropology, behavioral economics, evolutionary psychology, and philosophy, Herzog carefully crafts a seamless narrative enriched with real-life anecdotes, scientific research, and his own sense of moral ambivalence.
          </p>
          <p>
            Alternately poignant, challenging, and laugh-out-loud funny, this enlightening and provocative book will forever change the way we look at our relationships with other creatures and, ultimately, how we see ourselves.
          </p>
        </div>
      </div>
    </article>
  </div>
</div>
`;
